lib/archive/util.go

```diff
@@ -106,25 +120,26 @@ func AddMountInclusionsExclusions(currentMount string, filter *FilterSpec, mount
 	for _, mount := range mounts {
 		if strings.HasPrefix(mount, currentMount) && currentMount != mount {
 			// exclusions are relative to the mount so the leading `/` should be removed unless we decide otherwise.
-			exclusion := removeLeadingSlash(strings.TrimPrefix(mount, currentMount))
-			if !strings.HasSuffix(exclusion, "/") {
-				exclusion = exclusion + "/"
-			}
-
+			exclusion := clean(strings.TrimPrefix(mount, currentMount), true)
+			exclusion = strings.TrimPrefix(exclusion, "/") + "/"
 			filter.Exclusions[exclusion] = struct{}{}
 		}
 	}
 	return nil
 }
 
-// removeLeadingSlash will remove the '/' from in front of a target path
-// we use this to ensure relative pathing
-func removeLeadingSlash(path string) string {
-	return strings.TrimPrefix(path, "/")
-}
+// clean run filepath.clean on the target and will remove leading
+// and trailing slashes from the target path corresponding to supplied booleans
+func clean(path string, leading bool) string {
+	path = filepath.Clean(path)
+	// path returns '.' if the result of the clean was an empty string.
+	if path == "." {
+		return ""
+	}
+
+	if leading {
```

Comment:
Is wondering if the path is "/", with with leading is set true, the clean method will return "", is that expected change here?